catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/SourcePollerRunner.java

```diff
@@ -159,7 +159,14 @@ public void unbind(Source source) {
         LOGGER.debug("Unbinding source [{}]", source);
         if (source != null) {
             sources.remove(source);
-            sourceStatusThreadLocks.remove(cachedSources.remove(getSourceKey(source)));
+            SourceKey key = getSourceKey(source);
+            CachedSource cachedSource = null;
+            if(key != null){
```

Comment:
✏️  this always evaluates to `true` since getSourceKey() returns a new SourceKey. You could remove this condition.